File: contests_data/10/цветочный_магазин/solutions/time-limit-exceeded-or-accepted.py
n, q = map(int, input().split())
a = list(map(int, input().split()))
b = list(map(int, input().split()))
x = list(map(int, input().split()))
for i in range(q):
    c = a.copy()
    left = x[i]
    cnt = 0
    while True:
        ok = True
        for j in range(n):
            if c[j] >= b[j]:
                c[j] -= b[j]
            else:
                need = b[j] - c[j]
                if left >= need:
                    left -= need
                    c[j] = 0
                else:
                    ok = False
                    break
        if not ok:
            break
        cnt += 1
    print(cnt, end=" ")

File: contests_data/10/цветочный_магазин/solutions/time-limit-exceeded-or-accepted.cpp
#include <bits/stdc++.h>

#define int long long
using namespace std;

signed main() {
    int n, q;
    cin >> n >> q;
    vector<int> a(n);
    vector<int> b(n);
    vector<int> x(q);

    for (int i = 0; i < n; ++i) {
        cin >> a[i];
    }
    for (int i = 0; i < n; ++i) {
        b[i] = 1;
    }
    for (int i = 0; i < q; ++i) {
        cin >> x[i];
    }
    for (int i = 0; i < q; ++i) {
        vector<int> c(n);
        for (int j = 0; j < n; ++j) {
            c[j] = a[j];
        }
        int left = x[i];
        int cnt = 0;
        while (true) {
            bool ok = true;
            for (int j = 0; j < n; ++j) {
                if (c[j] >= b[j]) {
                    c[j] -= b[j];
                } else {
                    int need = b[j] - c[j];
                    if (left >= need) {
                        left -= need;
                        c[j] = 0;
                    } else {
                        ok = false;
                        break;
                    }
                }
            }
            if (!ok) {
                break;
            }
            cnt++;
        }
        cout << cnt << " ";
    }
}
